### PR TITLE
Setuptools integration / package migration

### PR DESCRIPTION
**Pull request by SamuelMarks**
Attempt #1 

Need more info on usage, maybe you can write up an argparse?

### CODE REVIEW

**Comment by abesto** [comment]
Looks good at a glance, will review in more detail and merge later.

Quick question: looking at https://github.com/wildducktheories/y2j I'm inclined to deprecate this project and direct people to `y2j`, their `yq` is by definition correct and feature-complete (it calls out to `jq`). You obviously put in a lot of effort into this PR, which to me means this `yq` has value for you over that `yq`. Can you summarize what that is?

**Comment by SamuelMarks** [comment]
You had a better description! - ahaha

I just used [my package generator](https://github.com/SamuelMarks/python-package-gen) on this then merged it together. So not a lot of effort. I did try `y2j`, but didn't get it to work the same way as `jq`, even for the simplest `cat foo.yml | yq .` case.

**Comment by abesto** [comment]
Heh, fair enough :) I'll update the README to point out `y2j`, and merge / publish this in case someone needs a pure python implementation for any reason.

**Comment by abesto** [comment]
Merged, released, worked like a charm, thanks!
